### DISCUSSION TITLE
rustacean and mason conflict?

### DISCUSSION THREAD

**Discussion by santoso-wijaya** [LazyVim/LazyVim#3549] [General]
I'm getting the following popup warning when I open a Rust file:

```
   Warn  21:50:39 notify.warn nvim-lspconfig.rust_analyzer has been setup.
This will likely lead to conflicts with the rustaceanvim LSP client.
See ':h rustaceanvim.mason'
```

Has anyone else seen it?

I think the same root cause is causing some doubling of overlay hints, e.g.:

![Screenshot 2024-06-08 at 10 37 33 PM](https://github.com/LazyVim/LazyVim/assets/157527329/d656388d-2b83-440f-aaf6-40ae3bc24f33)

**Comment by dpetka2001** [comment]
No, by default it doesn't. See at the [plugin's help files for more info](https://github.com/mrcjkb/rustaceanvim/blob/master/doc/mason.txt). 

You should install `rust-analyzer` via your [Rust toolchain](https://rust-analyzer.github.io/manual.html#rustup)

**Comment by santoso-wijaya** [comment]
I did already have rustup's `rust-analyzer` set up, I think. I don't currently have any lua file that sets up or overrides any settings from either mason or lspconfig, either.

I think I'll disable lspconfig's rust analyzer for now.

**Reply by dpetka2001** [reply]
Well LazyVim doesn't install `rust-analyzer` any more by default when you enable the Rust Extra. So, either you must have had it installed by an older LazyVim version or you installed it yourself.

**Reply by santoso-wijaya** [reply]
How do I find if that is the case? Running `whereis rust-analyzer` gives me only 1 entry--the `rustup`-installed one at `~/.cargo/bin/rust-analyzer`.

**Comment by latipun7** [comment]
- Uninstall your `rust-analyzer` from mason (<kbd>\<leader\>cm</kbd>, then <kbd>X</kbd> on `rust-analyzer`)
- Install `rust-analyzer` from rustup: `rustup component add rust-analyzer`

**Reply by santoso-wijaya** [reply]
That did the trick, thank you!

**Reply by silverjam** [reply]
Thanks @latipun7!

**Reply by xycool** [reply]
Thanks @latipun7 , I solved the problem 👍

**Comment by twilwa** [comment]
To clarify -- should I leave rust-analyzer uninstalled in Mason if I install the component via rustup? Sorry, new to things.

**Reply by dpetka2001** [reply]
Yes, there's no need for that, as `rustaceanvim` does not use `rust-analyzer` from Mason either way.
